src/cascadia/TerminalSettingsEditor/Profiles.xaml

```diff
@@ -59,41 +59,51 @@ the MIT License. See LICENSE in the project root for license information. -->
             <PivotItem x:Uid="Profile_General">
                 <ScrollViewer>
                     <StackPanel Style="{StaticResource PivotStackStyle}">
-                        
+
                         <!--Name-->
                         <ContentPresenter Visibility="{x:Bind State.Profile.IsBaseLayer, Mode=OneWay, Converter={StaticResource InvertedBooleanToVisibilityConverter}}"
                                           Style="{StaticResource SettingContainerStyle}"
                                           Margin="0,0,0,24">
                             <TextBox x:Uid="Profile_Name"
+                                     x:Name="ProfileName"
                                      Text="{x:Bind State.Profile.Name, Mode=TwoWay}"
+                                     AutomationProperties.HelpText="{Binding ProfileName.ToolTip, Mode=OneWay}"
                                      Style="{StaticResource TextBoxSettingStyle}"/>
                         </ContentPresenter>
 
                         <!--Commandline-->
-                        <ContentPresenter Visibility="{x:Bind State.Profile.IsBaseLayer, Mode=OneWay, Converter={StaticResource InvertedBooleanToVisibilityConverter}}"
-                                          Style="{StaticResource SettingContainerStyle}"
-                                          Margin="0,0,0,24">
+                        <local:SettingContainer x:Uid="Profile_Commandline"
+                                                x:Name="CommandlineContainer"
+                                                HasSettingValue="{x:Bind State.Profile.HasCommandline, Mode=OneWay}"
+                                                ClearSettingValue="{x:Bind State.Profile.ClearCommandline}"
+                                                Visibility="{x:Bind State.Profile.IsBaseLayer, Mode=OneWay, Converter={StaticResource InvertedBooleanToVisibilityConverter}}"
+                                                Margin="0,0,0,24">
                             <StackPanel Orientation="Horizontal">
-                                <TextBox x:Uid="Profile_Commandline"
-                                         x:Name="Commandline"
+                                <TextBox x:Name="Commandline"
                                          Text="{x:Bind State.Profile.Commandline, Mode=TwoWay}"
+                                         AutomationProperties.Name="{Binding CommandlineContainer.Header}"
+                                         AutomationProperties.HelpText="{Binding CommandlineContainer.HelpText}"
                                          Style="{StaticResource TextBoxSettingStyle}"/>

```

Comment:
This will be made unnecessary by the fixes in the SettingsContainer, right?

Reply:
Yup! Verified that it's no longer needed